polyfill/intersection-observer-test.js

```diff
@@ -3022,6 +3077,31 @@ describe('IntersectionObserver', function() {
           }
         ], done);
       });
+
+      it('handles tracking iframe viewport', function(done) {
+        iframe.style.height = '100px';
+        iframe.style.top = '100px';
+        iframeWin.scrollTo(0, 110);
+        // {root:iframeDoc} means to track the iframe viewport.
+        var io = createObserver(
+          function (records) {
+            io.unobserve(iframeTargetEl1);
+            var intersectionRect = rect({
+              top: 0, // if root=null, then this would be 100.
+              left: 0,
+              height: 90,
+              width: bodyWidth
+            })
+            expect(records.length).to.be(1);
+            expect(rect(records[0].rootBounds)).to.eql(getRootRect(iframeDoc));
+            expect(rect(records[0].intersectionRect)).to.eql(intersectionRect);
+            done();
+          },
+          { root: iframeDoc }
```

Comment:
I think we also need tests when `rootMargin` is involved.

Reply:
one for both cross-origin and same-origin? or is only same-origin fine?

Reply:
I think we need both same and cross-origin. But if that makes any difference, we probably need a pretty simple cross-origin and more detailed same-origin.

Reply:
Done